api-server/src/server/component-passport.js

```diff
@@ -118,10 +118,31 @@ export const createPassportCallbackAuthenticator =
           return next(err);
         }
 
-        if (!user || !userInfo) {
+        if (!user && !userInfo) {
           return res.redirect('/signin');
         }
 
+        const state = req && req.query && req.query.state;
+        // returnTo, origin and pathPrefix are audited by getReturnTo
+        let { returnTo, origin, pathPrefix } = getReturnTo(state, jwtSecret);
+        const redirectBase = getPrefixedLandingPath(origin, pathPrefix);
+
+        if (!user && userInfo == 'access_denied') {
+          const { error_description } = req.query;
+          const blockedByLaw =
+            error_description === 'Access denied from your location'
+              ? true
+              : false;
```

Comment:
```suggestion
          const blockedByLaw =
            error_description === 'Access denied from your location';
```

Reply:
Something to think about: Are we positive these description values will not change in patches/minors?

I would not have thought the only method provided by this API to perform this sort of comparison would be a case-sensitive string.